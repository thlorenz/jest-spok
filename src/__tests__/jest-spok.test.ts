import matchers from '../jest-spok'
import spok from 'spok'

spok.color = false
expect.extend(matchers)

describe('simple objects', () => {
  it('failing', () =>
    expect(() =>
      expect({ foo: 1, bar: {}, baz: 3 }).toSatisfy({
        foo: spok.ge(2),
        bar: spok.string,
        baz: 3,
      })
    ).toThrowErrorMatchingSnapshot())
  it('passing', () =>
    expect({ foo: 2, bar: 'hello' }).toSatisfy({
      foo: spok.ge(2),
      bar: spok.string,
    }))
})

describe('nested objects', () => {
  it('failing', () =>
    expect(() =>
      expect({
        uno: { eins: 1, one: 1, unoDeep: { elf: 11, eleven: 11 } },
      }).toSatisfy({
        uno: {
          eins: spok.gez,
          one: spok.lez,
          unoDeep: { elf: spok.function, eleven: 11 },
        },
      })
    ).toThrowErrorMatchingSnapshot())
})

describe('arrays', () => {
  it('failing', () =>
    expect(() =>
      expect([{ foo: 1 }, { bar: 2 }, { baz: 3 }]).toSatisfy([
        { foo: spok.ge(1) },
        { bar: spok.le(1) },
        { baz: spok.string },
      ])
    ).toThrowErrorMatchingSnapshot())
})
