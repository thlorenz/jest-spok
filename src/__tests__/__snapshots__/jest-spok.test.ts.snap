// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrays failing 1`] = `
"expect(received).toSatisfy(expected)

Mismatches:
  \\"Â·Â· bar = 2  [90msatisfies: spok.le(1)[39m\\"
  \\"Â·Â· baz = 3  [90msatisfies: spok.string[39m\\"

Received:
  [{\\"foo\\": 1}, {\\"bar\\": 2}, {\\"baz\\": 3}]"
`;

exports[`nested objects failing 1`] = `
"expect(received).toSatisfy(expected)

Mismatches:
  \\"Â·Â· one = 1  [90msatisfies: spok.lez[39m\\"
  \\"Â·Â· Â·Â· elf = 11  [90msatisfies: spok.function[39m\\"

Received:
  {\\"uno\\": {\\"eins\\": 1, \\"one\\": 1, \\"unoDeep\\": {\\"eleven\\": 11, \\"elf\\": 11}}}"
`;

exports[`no specs prints helping message 1`] = `
"expect(received).toSatisfy(expected)

No Specifications provided, adapt your code as follows and edit the specs as needed:

  expect(actual)
    .toSatisfy( [ { foo: 1 }, { bar: 2 }, { baz: 3 } ])
"
`;

exports[`simple objects failing 1`] = `
"expect(received).toSatisfy(expected)

Mismatches:
  \\"foo = 1  [90msatisfies: spok.ge(2)[39m\\"
  \\"bar = {}  [90msatisfies: spok.string[39m\\"

Received:
  {\\"bar\\": {}, \\"baz\\": 3, \\"foo\\": 1}"
`;
